embassy-net/src/lib.rs

```diff
@@ -45,25 +47,16 @@ use crate::device::DriverAdapter;
 const LOCAL_PORT_MIN: u16 = 1025;
 const LOCAL_PORT_MAX: u16 = 65535;
 
-pub struct StackResources<const ADDR: usize, const SOCK: usize, const NEIGHBOR: usize> {
-    addresses: [IpCidr; ADDR],
+pub struct StackResources<const SOCK: usize, const NEIGHBOR: usize> {
+    addresses: [IpCidr; 5],
```

Comment:
`addresses` is now owned by the smoltcp iface, there's no need to have it in this struct (smoltcp iface borrowed it before)